w3-revocations-check.md

```diff
@@ -0,0 +1,185 @@
+- **Encoding:** CID will be normalized to string format internally
+
+### Response Format
+
+#### Success Response (200): Delegation Revoked
+
+**Content-Type:** `application/vnd.ipld.car`
+
+Returns a CAR (Content Addressable aRchive) file containing:
+
+- Revocation metadata in `revocations@0.0.1` schema
+- Embedded UCAN proof blocks for trustless verification
+- ETag header for caching optimization
+
+##### CAR File Structure
+
+The CAR file contains a root block with the following structure:
+
+```json
+{
+  "revocations@0.0.1": {
+    "revocations": [
+      {
+        "delegation": { "/": "[delegationCID]" },
+        "scope": "[scopeDID]", 
+        "cause": { "/": "[causeCID]" }
+      }
+    ]
+  }
+}
+```
+
+##### Embedded Proof Blocks
+
+The CAR file includes the complete UCAN revocation proof referenced by `causeCID`, enabling clients to:
+
+- Cryptographically verify the revocation signature
+- Validate the revocation authority chain
+- Perform trustless verification without server dependency
+
+#### Not Found Response (404): Delegation Not Revoked
+
+**Content-Type:** `text/plain`
+
+```text
+Delegation not revoked
+```
+
+This indicates the delegation CID has not been explicitly revoked. Clients should check the delegation's proof chain to determine if it's invalid due to revoked dependencies.
+
+### Error Responses
+
+#### 400 Bad Request
+
+**Content-Type:** `application/json`
+
+##### Invalid CID Parameter
+
+```json
+{
+  "error": "Bad request",
+  "message": "Invalid CID parameter"
+}
+```
+
+##### Missing CID Parameter
+
+```json
+{
+  "error": "Bad request", 
+  "message": "CID parameter is required"
+}
+```
+
+#### 500 Internal Server Error
+
+**Content-Type:** `application/json`
+
+##### Query Failure
+
+```json
+{
+  "error": "Internal server error",
+  "message": "Failed to query revocations"
+}
+```
+
+##### General Error
+
+```json
+{
+  "error": "Internal server error",
+  "message": "An unexpected error occurred"
+}
+```
+
+### CORS Headers
+
+The endpoint includes CORS headers for cross-origin requests:
+
+- `Access-Control-Allow-Origin: *`
+- `Access-Control-Allow-Methods: GET`
+- `Access-Control-Allow-Headers: Accept`
+
+# Usage Examples
+
+## cURL Example
+
+```bash
+# Check if a delegation is revoked
+curl -X GET https://up.storacha.network/revocations/{cid}
+
+# Save CAR file for offline verification (when response is 200)
+curl -X GET https://up.storacha.network/revocations/{cid} \
+  -H "Accept: application/vnd.ipld.car" \
+  -o revocation-proof.car
+```
+
+## Client-Side Proof Chain Verification
+
+```javascript
+/**
+ * Recursively verify the delegation proof chain for revocations
+ */
+async function verifyProofChain(delegation) {
+  // Check if delegation is explicitly revoked
+  const response = await fetch(`https://up.storacha.network/revocations/${delegation.cid}`);
+  
+  if (response.status === 200) {
+    return { isValid: false, reason: 'Delegation explicitly revoked' };
```

Comment:
Note: this is not doing verification, it's _trusting_ the 200 from the service.

Would be worth building a client library that does the verification.

I would perhaps check from the start of the delegation chain first.